xc/cache.lua

```diff
@@ -35,9 +35,10 @@ end
 -- @return true if authorized, false if denied, nil if unknown
 -- @return reason why the authorization is denied (optional)
 function _M.authorize(service_id, credentials, usage_method)
-  local redis, ok = redis_pool.acquire()
+  local redis, ok, err = redis_pool.acquire()
 
   if not ok then
+    ngx.log(ngx.WARN, "[cache] couldn't connect to redis: ", err)
```

Comment:
Also, we should add a test that checks that we log a warning when there's an error connecting to Redis.
That test should be under this describe clause: `describe('when there is an error acquiring a Redis connection', function()` in `cache_spec.lua`.